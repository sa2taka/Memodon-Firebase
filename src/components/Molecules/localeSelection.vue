<template>
  <v-menu>
    <template v-slot:activator="{ on }">
      <v-btn text color="primary" v-on="on">
        <i :class="`flag-icon flag-icon-${selected}`"></i>
      </v-btn>
    </template>
    <v-list>
      <v-list-item v-for="name in nations" :key="name" @click="setLang(name)">
        <i :class="`flag-icon flag-icon-${name}`"></i>
      </v-list-item>
    </v-list>
  </v-menu>
</template>

<script lang="ts">
import { Component, Watch, Vue } from 'vue-property-decorator';

@Component
export default class LocaleSelection extends Vue {
  // lang: nation name
  public nations = ['jp', 'us'];
  public selected = 'en';

  public created() {
    this.selected = this.$i18n.locale;
  }

  public setLang(locale: string) {
    this.selected = locale;
    this.$i18n.locale = locale;
  }
}
</script>

<style lang="scss" scoped></style>
