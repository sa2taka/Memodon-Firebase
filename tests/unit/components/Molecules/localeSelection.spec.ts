import { shallowMount, createLocalVue } from '@vue/test-utils';
import localeSelection from '@/components/Molecules/localeSelection.vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import i18n from '@/i18n';
import VueI18n from 'vue-i18n';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(Vuetify);
localVue.use(VueI18n);

describe('Component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(localeSelection, { i18n, localVue });
  });

  it('is a Vue instance', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should set locale', () => {
    wrapper.vm.setLang('jp');
    expect(wrapper.vm.$i18n.locale).toBe('jp');
  });
});
