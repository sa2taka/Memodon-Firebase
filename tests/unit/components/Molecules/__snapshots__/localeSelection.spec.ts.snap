// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is a Vue instance 1`] = `
<v-menu-stub
  closedelay="0"
  closeonclick="true"
  closeoncontentclick="true"
  contentclass=""
  maxheight="auto"
  maxwidth="auto"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  opendelay="0"
  openonclick="true"
  origin="top left"
  transition="v-menu-transition"
>
   
  <v-list-stub
    tag="div"
    tile="true"
  >
    <v-list-item-stub
      activeclass=""
      tag="div"
    >
      <i
        class="flag-icon flag-icon-jp"
      />
    </v-list-item-stub>
    <v-list-item-stub
      activeclass=""
      tag="div"
    >
      <i
        class="flag-icon flag-icon-us"
      />
    </v-list-item-stub>
  </v-list-stub>
</v-menu-stub>
`;
